pkg/core/core.go

```diff
@@ -222,6 +222,8 @@ func (c *Core) Run(ctx context.Context, id uint64, opts models.RunOptions) model
 		if appErr.Err != nil {
 			utils.LogError(c.logger, appErr, "error while running the app")
 			appErrCh <- appErr
+		} else if appErr.AppErrorType == models.ErrCtxCanceled {
```

Comment:
I think checking here is a bit late. Because for some apps it takes some time to close . We can actually check on the context which this function is using and also when app itself has an error then also we can do that. So there will be two checks 
1. On the context of this function once context is cancelled we should signal proxy and 
2. When application stops and if it is not context err we need to send signal to proxy. Why we ar ignoring context cancel here is that if it is context cancel we will already know it in the first case.